### PR TITLE
Refreshing the lib slightly.

### PR DESCRIPTION
**Pull request by Narsil**

### CODE REVIEW

File: src/lib.rs
Comment (McPatate) [inline src/lib.rs:1]: You could use thiserror as well, code would be a bit leaner but I'm ok with it as is

File: .github/workflows/test.yml
Comment (McPatate) [inline .github/workflows/test.yml:1]: Nice

**Review by McPatate — APPROVED** [review]
Lgtm, ty!

File: src/lib.rs
Comment (Narsil) [reply src/lib.rs:1]: Avoiding adding a dependency for this overly simple one. Definitely let's add it if it every becomes much more verbose.

**Pull request merged**